deploy/swrs/public/view/carbon_tax_calculation.sql

```diff
@@ -8,86 +8,109 @@ begin;
-                      on _fuel.report_id = _report.id
-                 join swrs.organisation as _organisation
-                      on _report.id = _organisation.report_id
-                 left join swrs.facility as _facility
-                      on _report.id = _facility.report_id
-                 left join swrs.naics as _naics
-                      on _report.id = _naics.report_id
-                      and ((_naics.path_context = 'RegistrationData'
-                      and (_naics.naics_priority = 'Primary'
-                            or _naics.naics_priority = '100.00'
-                            or _naics.naics_priority = '100')
-                      and (select count(eccc_xml_file_id)
-                           from swrs.naics as __naics
-                           where eccc_xml_file_id = _emission.eccc_xml_file_id
-                           and __naics.path_context = 'RegistrationData'
-                           and (__naics.naics_priority = 'Primary'
-                            or __naics.naics_priority = '100.00'
-                            or __naics.naics_priority = '100')) < 2)
-                       or (_naics.path_context='VerifyTombstone'
-                           and _naics.naics_code is not null
-                           and (select count(eccc_xml_file_id)
-                           from swrs.naics as __naics
-                           where eccc_xml_file_id = _emission.eccc_xml_file_id
-                           and __naics.path_context = 'RegistrationData'
-                           and (__naics.naics_priority = 'Primary'
-                            or __naics.naics_priority = '100.00'
-                            or __naics.naics_priority = '100')) > 1))
-                 join swrs.activity as _activity
-                      on _unit.activity_id = _activity.id
-                 join swrs.fuel_carbon_tax_details as _fuel_carbon_tax_details
-                      on _fuel_mapping.fuel_carbon_tax_details_id = _fuel_carbon_tax_details.id
-                 join swrs.carbon_tax_act_fuel_type as _cta
-                      on _fuel_carbon_tax_details.carbon_tax_act_fuel_type_id = _cta.id
-                 join swrs.fuel_charge as _fuel_charge
-                      on _fuel_charge.carbon_tax_act_fuel_type_id = _cta.id
-                      and (concat(_report.reporting_period_duration::text, '-12-31')::date
-                      between _fuel_charge.start_date and _fuel_charge.end_date)
+  with fuel_data as (
+
+    -- FLARED EMISSION
+    select
+      e.report_id,
+      'Natural Gas (Sm^3)' as fuel_type,
+      e.emission_category,
+      e.fuel_mapping_id,
+      fm.fuel_carbon_tax_details_id,
+      ctd.carbon_tax_act_fuel_type_id,
+      (coalesce(e.quantity, 0::numeric) * ctd.unit_conversion_factor::numeric) as fuel_amount
+    from swrs.emission e
+    join swrs.fuel_mapping fm
+      on e.fuel_mapping_id = fm.id
+      and e.fuel_mapping_id = 145 -- Flared Natural Gas CO2
```

Comment:
```suggestion
      and e.fuel_mapping_id = (select id from swrs.fuel where fuel_name = 'Flared Natural Gas CO2')
```
This would be a good example of the "no magic numbers" rule